src/implementations/astroport/stableswap.rs

```diff
@@ -0,0 +1,190 @@
+use astroport_core::querier::query_supply;
+use astroport_core::U256;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Decimal, Env, MessageInfo, Response, StdError, StdResult};
+use cosmwasm_std::{Deps, Uint128};
+use cw_asset::{Asset, AssetInfo, AssetInfoBase, AssetList};
+use delegate::delegate;
+use std::cmp::Ordering;
+
+use crate::traits::Pool;
+use crate::CwDexError;
+use astroport_core::asset::Asset as AstroAsset;
+use astroport_core::asset::AssetInfo as AstroAssetInfo;
+
+use super::base_pool::AstroportBasePool;
+use super::helpers::{compute_current_amp, compute_d, AstroAssetList};
+use super::querier::{query_asset_precision, query_pair_config};
+
+pub const N_COINS: u8 = 2;
+
+#[cw_serde]
+pub struct AstroportStableSwapPool(AstroportBasePool);
+
+impl AstroportStableSwapPool {
+    pub fn new(pair_addr: Addr, lp_token_addr: Addr) -> Self {
+        Self(AstroportBasePool {
+            pair_addr,
+            lp_token_addr,
+        })
+    }
+}
+
+impl Pool for AstroportStableSwapPool {
+    // Delegate all methods except `simulate_provide_liquidity` to the AstroportBasePool implementations
+    delegate!(
+        to self.0 {
+            fn get_pool_liquidity(&self, deps: Deps) -> Result<AssetList, CwDexError>;
+            fn simulate_withdraw_liquidity(&self, deps: Deps, asset: Asset) -> Result<AssetList, CwDexError>;
+            fn simulate_swap(&self, deps: Deps, offer_asset: Asset, ask_asset_info: AssetInfo, sender: Option<String>) -> StdResult<Uint128>;
+            fn provide_liquidity(&self, deps: Deps, env: &Env, info: &MessageInfo, assets: AssetList, slippage_tolerance: Option<Decimal>) -> Result<Response, CwDexError>;
+            fn withdraw_liquidity(&self, deps: Deps, env: &Env, asset: Asset) -> Result<Response, CwDexError>;
+            fn swap(&self, deps: Deps, env: &Env, offer_asset: Asset, ask_asset_info: AssetInfo, minimum_out_amount: Uint128,) -> Result<Response, CwDexError>;
+        }
+    );
+
+    fn simulate_provide_liquidity(
+        &self,
+        deps: Deps,
+        env: Env,
+        assets: AssetList,
+    ) -> Result<Asset, CwDexError> {
+        let assets: AstroAssetList = assets.to_owned().try_into()?;
+        let config = query_pair_config(&deps.querier, &self.0.pair_addr)?;
+        let mut pools: [AstroAsset; 2] =
+            config.pair_info.query_pools(&deps.querier, env.contract.address.clone())?;
```

Comment:
```rust
/// Returns the balance for each asset in the pool.
/// ## Params
/// * **self** is the type of the caller object
///
/// * **querier** is an object of type [`QuerierWrapper`]
///
/// * **contract_addr** is pair's pool address.
pub fn query_pools(
```
from the astro code

So `env.contract.address` is not going to be correct here. Should use `self.0.pair_addr` instead